lib/ts/ink_res_init.cc

```diff
@@ -503,7 +503,20 @@ ink_res_init(ink_res_state statp,         ///< State object to update.
           hints.ai_family = PF_UNSPEC;
           hints.ai_socktype = SOCK_DGRAM; /*dummy*/
           hints.ai_flags = AI_NUMERICHOST;
-          sprintf(sbuf, "%d", NAMESERVER_PORT);
+
```

Comment:
You should use `ats_ip_parse` here. It was made specifically to handle this kind of task.